docs/platform_developers/developer_getting_started.rst

```diff
@@ -0,0 +1,154 @@
+
+At the time of this writing, linting and code navigation in IDEs work only if you have a project, so even if you only work on a single module, it is best to have a project.
+
+There are two scenarios:
+
+1. There is already an existing project that you can ``git clone``.
+2. Make a new project.
+
+Steps to create a project are mentioned `here <https://docs.inmanta.com/community/latest/model_developers/configurationmodel.html>`_ for further reading.
+
+``cookiecutter`` can be used to create projects in an easier and convenient fashion. It could be utilized like:
+
+.. code-block:: bash
+
+    pip install cookiecutter
+
+    cookiecutter https://github.com/inmanta/inmanta-project-template.git
+
+Further information about cookiecutter can be found `here <https://github.com/inmanta/inmanta-project-template>`_ and `here <https://docs.inmanta.com/community/latest/model_developers/configurationmodel.html>`_.
+
+
+* If you are working on an existing project, they come with ``requirements.txt`` or ``requirements.dev.txt`` to install the required modules:
+
+.. code-block:: bash
+
+    pip install -r requirements.txt
+
+    pip install -r requirements.dev.txt
+
+* If you are working on a new project, you need to install some essential packages as follows:
+
+.. code-block:: bash
+
+    pip install inmanta-core
+
+    pip install pytest
+
+    pip install pytest-inmanta
+
+
+Once you are done with creating a project and installing the required modules, you can ``cd`` into that directory and open vs code by running:
+
+.. code-block:: bash
+    
+    cd <project_name>
+    
+    code .
+
+Upon opening your vs code, and the ``main.cf`` file, you should see modules downloading in ``libs`` directory.
+
+
+Set project sources
+#####################
+
+When starting a new project, the next step is to set the sources of your project so that it knows, where to get its required modules from. Otherwise, you can skip this step and just ``import`` your desired modules.
+
+If you only use opensource modules as provided by Inmanta, you can skip below step. 
+
+1. Find the module you want to work on
+2. Copy the SSH URL of the repo
+3. In your VS code, open the ``project.yml`` file and under ``repo:``, add the copied line there but keep in mind to replace the name of a specific module with a place holder, like below example:
+
+.. code-block:: bash
+
+    code project.yml
+
+.. code-block:: yaml
+    
+    repo:
+        - git@code.inmanta.com:example/my_module.git
+
+Becomes:
+
+.. code-block:: yaml
+    
+    repo:
+        - git@code.inmanta.com:example/{}.git
+
+* Now, in your ``main.cf`` file, if you import a module like, ``import <my_module>`` and save the file, you can get code completion. If you are working on an exisitng project with a populated ``main.``cf file, code completion will work as expected.
+
+**Please note, code completion and navigation work on modules that are imported in the ``main.cf`` file**.
+
+
+Module developers guide
```

Comment:
I would add a section to clone or create a module, similar to the one about the project.

Reply:
Done

Reply:
Best to also maintain the split between existing module and new module